test/test_torch.py

```diff
@@ -7045,6 +7045,41 @@ def test_cat_out_channels_last(self, device):
         res2 = torch.cat((x, y), out=z)
         self.assertEqual(res1, res2)
 
+    # This test ensures that all CUDA implementations of operations that use the
+    # non-deterministic `atomicAdd` function raise an error when
+    # `torch.get_deterministic() == True`
+    @onlyCUDA
+    def test_not_deterministic_alert_atomicAdd(self, device):
```

Comment:
It seemed like a good idea to add a test to make sure that the alert is working for the operations I've added it to. But I didn't want to add cases for everything yet because I'd like to get y'all's opinions about how I set up the test first.

Reply:
It might be easier to test for the alerts by the existing `test_common_methods` mechanism, as those test cases will already have valid input sizes to run your operation on. You can just add a decorator that says "assert that a nondet warning was raised".

Reply:
That should work for most cases, but potentially not for those than have non-determinism only in some branch (in which might be good to also test that the deterministic cases don't raise).

Reply:
@ezyang, could you point me to what you're talking about? I searched the repo for `test_common_methods` and nothing came up.

Another idea: what if I implement these tests in C++? That way I can just call backward functions directly instead of having to call the forward first.

Reply:
That's the file. There's also some related concepts in other files.

> Another idea: what if I implement these tests in C++?

It's a lot more annoying to test for warnings in C++... I'd probably rather just bind the backward functions from Python (in some private API) and call them that way, in that case...

Reply:
If you want to try the C++ test route I'm not opposed, though.